### ISSUE TITLE
[Bug] `defineProps`/`defineEmits` is not defined

### ISSUE THREAD

**Issue by wenfangdu** [vuejs/eslint-plugin-vue#1597]
**Checklist**

- [x] I have tried restarting my IDE and the issue persists.
- [x] I have read the [FAQ](https://eslint.vuejs.org/user-guide/#faq) and my problem is not listed.

**Tell us about your environment**

- **ESLint version:** 7.32.0
- **eslint-plugin-vue version:** 7.15.1
- **Node version:** 16.5.0
- **Operating System:** Windows 10

**What did you do**

```vue
<script setup>
  defineProps()

  defineEmits()
</script>
```

**What did you expect to happen?**

ESLint `globals` are configured automatically like the following by `eslint-plugin-vue` and no error is thrown:

```json
{
  "globals": {
    "defineProps": "readonly",
    "defineEmits": "readonly"
  }
}
```

**What actually happened?**

```sh
'defineProps' is not defined. eslint(no-undef)

'defineEmits' is not defined. eslint(no-undef)
```

**Comment by ota-meshi** [comment]
I can't reproduce your problem. Share the smallest repository that can reproduce your problem.

**Comment by ota-meshi** [comment]
I misunderstood your problem.

> ESLint globals are configured automatically

It is not automatically configured. Add globals in your configuration file. Please read the FAQ.
https://eslint.vuejs.org/user-guide/#compiler-macros-such-as-defineprops-and-defineemits-are-warned-by-no-undef-rule

**Comment by zds-d** [comment]
`const emit = defineEmits(['formSubmit'])` cannot be written in a function ，If it is written in an arrow function, it will also report an error: defineEmits is not defined
